registry/storage/driver/s3-aws/s3.go

```diff
@@ -942,15 +942,152 @@ func (d *driver) Walk(ctx context.Context, from string, f storagedriver.WalkFn)
 	return nil
 }
 
-func (d *driver) doWalk(parentCtx context.Context, objectCount *int64, path, prefix string, f storagedriver.WalkFn) error {
+func (d *driver) doWalk(parentCtx context.Context, objectCount *int64, path, prefix string, f storagedriver.WalkFn, opts ...storagedriver.WalkOpt) error {
+	options := storagedriver.ToWalkOptions(opts...)
+	if options.StopOnErrSkipDir {
+		return d.doWalkWithoutSkipDir(parentCtx, objectCount, path, prefix, f)
+	}
+	_, err := d.doWalkRecursive(parentCtx, objectCount, path, prefix, f)
+	return err
+}
+
+type walkInfoContainer struct {
```

Comment:
I'd prefer to call this simply `walkInfo`; also, I'm wondering if we need the getter funcs for this type. Is there any specific reason we are doing that here? (my suspicion is it's merely to make the referenced field shorter in the code?)